src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts

```diff
@@ -134,33 +142,39 @@ describe('WdsDataTableProvider', () => {
     return Promise.resolve({ message: 'Upload Succeeded', recordsModified: 1 })
   }
 
-  const getV2AppInfoMockImpl: AppsContract['getV2AppInfo'] = (_workspaceId: string) => {
+  const listAppsV2MockImpl: AppsContract['listAppsV2'] = (_workspaceId: string) => {
     return Promise.resolve(testProxyUrlResponse)
   }
 
+  const createAppV2MockImpl: AppsContract['createAppV2'] = (_workspaceId: string) => {
+    return Promise.resolve('')
+  }
+
   let getRecords: jest.MockedFunction<WorkspaceDataContract['getRecords']>
   let deleteTable: jest.MockedFunction<WorkspaceDataContract['deleteTable']>
   let downloadTsv: jest.MockedFunction<WorkspaceDataContract['downloadTsv']>
   let uploadTsv: jest.MockedFunction<WorkspaceDataContract['uploadTsv']>
-  let getV2AppInfo: jest.MockedFunction<AppsContract['getV2AppInfo']>
+  let listAppsV2: jest.MockedFunction<AppsContract['listAppsV2']>
+  let createAppV2: jest.MockedFunction<AppsContract['createAppV2']>
 
   beforeEach(() => {
     getRecords = jest.fn().mockImplementation(getRecordsMockImpl)
     deleteTable = jest.fn().mockImplementation(deleteTableMockImpl)
     downloadTsv = jest.fn().mockImplementation(downloadTsvMockImpl)
     uploadTsv = jest.fn().mockImplementation(uploadTsvMockImpl)
-    getV2AppInfo = jest.fn().mockImplementation(getV2AppInfoMockImpl)
+    listAppsV2 = jest.fn().mockImplementation(listAppsV2MockImpl)
+    createAppV2 = jest.fn().mockImplementation(createAppV2MockImpl)
 
     asMockedFn(Ajax).mockImplementation(() => ({
       WorkspaceData: { getRecords, deleteTable, downloadTsv, uploadTsv } as Partial<WorkspaceDataContract>,
-      Apps: { getV2AppInfo } as Partial<AppsContract>
+      Apps: { listAppsV2, createAppV2 } as Partial<AppsContract>
     } as Partial<AjaxContract> as AjaxContract))
   })
 
   describe('transformAttributes', () => {
     it('excludes the primary key from the resultant attributes', () => {
       // Arrange
-      const provider = new TestableWdsProvider(uuid)
+      const provider = new TestableWdsProvider(uuid, false)
```

Comment:
the existing TestableWdsProvider pattern to make the two protected transform methods testable seems like a lot of hassle.... I probably would have just broken out the conversion logic to a pageConverter utility above the main class, export it for testability, and just don't export it at the package level when-and-if this code ever becomes cross-team sharable.  The tests here could then exercise the converter directly with much less hassle.... and the main class would still leverage that same code as part of the getPage method.

Reply:
Similar to https://github.com/DataBiosphere/terra-ui/pull/3709#discussion_r1091186347

Reply:
the `TestableWdsProvider` model is my bad, showing inexperience with TS and its private/protected/etc model

agree that refactoring this would be nice but should not block this PR.